source/Halibut/DataStream.cs

```diff
@@ -32,7 +32,16 @@ public DataStream(long length, Action<Stream> writer)
 
         public IDataStreamReceiver Receiver()
         {
-            return receiver ?? new InMemoryDataStreamReceiver(writer);
+            if (receiver != null)
+                return receiver;
+
+            // Use a FileStream for packages over 2GB, or you risk running into OutOfMemory
+            // exceptions with MemoryStream.
+            var maxMemoryStreamLength = 2147483648;
```

Comment:
Good call. I've switched this to Int32.MaxValue to avoid any potential issues on 64-bit systems